### PR TITLE
[3.3][Kernel] Disable setting columnMapping/icebergCompatV2 for 3.3.0

### PR DESCRIPTION
**Pull request by vkorukanti**
There are a couple of pieces missing from fully supporting these features
- Data path PR is not yet merged (delta-io/delta#3475), waiting on the expression fixup which is going to affect how column mapping is done.
- Auto enabling of the protocol version based on the current table version and column mapping/iceberg compat v2 enabled.

Disable these in 3.3 to avoid writing a partial column mapping data.

### CODE REVIEW

File: kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java [hunk 1]
```diff
@@ -254,6 +254,20 @@ public static Map<String, String> validateProperties(Map<String, String> configu
       String key = kv.getKey().toLowerCase(Locale.ROOT);
       String value = kv.getValue();
       if (key.startsWith("delta.") && VALID_PROPERTIES.containsKey(key)) {
+        // Disable `columnMapping` and `icebergCompatV2` for Delta 3.3 release
```
Comment (allisonport-db) [inline kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java:257]: Should we do this check after we filter out to only have the "new properties"? I'm not super familiar with the expected semantics with the table properties code on writes, but it seems like there might be existing properties here for a table that already has column mapping enabled?

File: kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableWritesSuite.scala [hunk 2]
```diff
@@ -1256,4 +1256,33 @@ class DeltaTableWritesSuite extends DeltaTableWriteSuiteBase with ParquetSuiteBa
...
+            tableProperties = Map(key -> value))
+            .commit(engine, emptyIterable())
+        }
+        assert(ex.getMessage.contains("Unknown configuration was specified:"))
+      }
+    }
+  }
+
+  // helper method to run columnmapping tests
+  def withColumnMappingEnabled(testName: String)(fun: => Unit): Unit = {
```
Comment (allisonport-db) [inline kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableWritesSuite.scala:1277]: maybe this should be with more specific to column mapping _enablement_ being enabled? confusing wording :(

File: kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java (hunk 1 above)
Comment (vkorukanti) [reply kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java:257]: Very good point. Just added a fix and test to make sure it is not writable for existing column mapping enabled tables.

File: kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableWritesSuite.scala (hunk 2 above)
Comment (vkorukanti) [reply kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableWritesSuite.scala:1277]: renamed.

**Review by allisonport-db — APPROVED** [review]

**Pull request merged**